src/plugins/memoryvalue/memoryvalue.c

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file
+ *
+ * @brief memoryvalue plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#define _XOPEN_SOURCE
+#include "memoryvalue.h"
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdbtypes.h>
+#include <regex.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+static kdb_unsigned_long_long_t is_valid_key (Key * key)
+{
+
+	const Key * meta = keyGetMeta (key, "check/memoryvalue");
```

Comment:
You already check that `check/memoryvalue` exists before calling this function and you don't use its value, so no need to check again.